gobblin-utility/src/main/java/gobblin/util/wiki/ConfigWikiGenerator.java

```diff
@@ -0,0 +1,189 @@
+
+  /**
+   * List of tables to blacklisted of the form [dbpattern.tablepattern1|tablepattern2|...]
+   */
+  @ConfigKey(required = false, def = "-", group = "Hive Distcp")
+  private static final String HIVE_DATASET_FINDER_BLACKLIST = "hive.dataset.blacklist";
+
+  private static final String DOCS_HTML_OUTPUT_PATH = "/tmp/configDocs.html";
+  private static final String DOCS_HTML_TEMPLATE_RESOURCE = "configDocs.ftl";
+
+  /**
+   * Invoked by the doclet task
+   */
+  public static boolean start(RootDoc root) {
+
+    Multimap<String, ConfigKeyObj> groupConfigKeys = ArrayListMultimap.create();
+    try {
+
+      for (ClassDoc c : root.classes()) {
+        // Currently only run for this class
+        //if (c.qualifiedTypeName().equals("gobblin.util.wiki.ConfigWikiGenerator")) {
+          for (FieldDoc f : c.fields(false)) {
+            for (AnnotationDesc annotationDesc : f.annotations()) {
+              if (annotationDesc.annotationType().qualifiedName().equals(ConfigKey.class.getName())) {
+
+                ConfigKeyObj obj = new ConfigKeyObj();
+                obj.setName(f.constantValue().toString());
+                obj.setDoc(f.commentText());
+                for (ElementValuePair elementValuePair : annotationDesc.elementValues()) {
+                  if (elementValuePair.element().qualifiedName().equals(ConfigKey.class.getName() + ".required")) {
+                    obj.setRequired(elementValuePair.value().toString());
+                  } else if (elementValuePair.element().qualifiedName().equals(ConfigKey.class.getName() + ".def")) {
+                    obj.setDef(StringUtils.remove(elementValuePair.value().toString(), "\""));
+                  } else if (elementValuePair.element().qualifiedName().equals(ConfigKey.class.getName() + ".group")) {
+                    obj.setGroup(StringUtils.remove(elementValuePair.value().toString(), "\""));
+                  }
+
+                }
+                groupConfigKeys.put(obj.getGroup(), obj);
+              }
+            }
+
+          }
+        //}
+      }
+
+      createHtml(groupConfigKeys, DOCS_HTML_OUTPUT_PATH);
+    } catch (Throwable t) {
+      System.out.println("Failed to create Config keys doc. Build will still succeed");
+      t.printStackTrace();
+    }
+
+    return true;
+  }
+
+  private static void createHtml(Multimap<String, ConfigKeyObj> groupConfigKeys, String htmlPath) throws Exception {
+
+    Configuration config = new Configuration();
+
+    config.setClassForTemplateLoading(ConfigWikiGenerator.class, "templates");
+    config.setDefaultEncoding("UTF-8");
+    config.setTemplateExceptionHandler(TemplateExceptionHandler.RETHROW_HANDLER);
+
+    Map<String, Object> input = new HashMap<String, Object>();
+
+    input.put("title", "Configuration Keys");
+    input.put("groupConfigKeys", new TreeMap<>(groupConfigKeys.asMap()));
+
+    Template template = config.getTemplate(DOCS_HTML_TEMPLATE_RESOURCE);
```

Comment:
would be nice to have a template for config file, that generates config file for a give source and its related tightly coupled components (such as extractor)
may be we can leverage grouping information for that or define it as an a new property